runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -0,0 +1,311 @@
+        K defaultKeyMessage = (K) kClass.getMethod("getDefaultInstance").invoke(null);
+        addTypeToClassMap(defaultKeyMessage);
+
+        V defaultValueMessage = (V) vClass.getMethod("getDefaultInstance").invoke(null);
+        addTypeToClassMap(defaultValueMessage);
+
+        M defaultMetadataMessage = null;
+        if (mClass != null) {
+            defaultMetadataMessage = (M) mClass.getMethod("getDefaultInstance").invoke(null);
+            addTypeToClassMap(defaultMetadataMessage);
+        }
+
+        String fullyQualifiedTableName = getFullyQualifiedTableName(namespace, tableName);
+
+        // Open and return table instance.
+        Table<K, V, M> table = new Table<>(
+                namespace,
+                fullyQualifiedTableName,
+                defaultValueMessage,
+                defaultMetadataMessage,
+                this.runtime,
+                this.protobufSerializer);
+        tableMap.put(fullyQualifiedTableName, (Table<Message, Message, Message>) table);
+
+        registerTable(namespace, tableName, kClass, vClass, mClass);
+        return table;
+    }
+
+    /**
+     * Get an already opened table. Fetches the table from the cache given only the namespace and table name.
+     * Throws a NoSuchElementException if table is not previously opened and not present in cache.
+     *
+     * @param namespace Namespace of the table.
+     * @param tableName Name of the table.
+     * @param <K>       Key type.
+     * @param <V>       Value type.
+     * @param <M>       Metadata type.
+     * @return Table instance.
+     */
+    public <K extends Message, V extends Message, M extends Message>
+    Table<K, V, M> getTable(String namespace, String tableName) {
+        String fullyQualifiedTableName = getFullyQualifiedTableName(namespace, tableName);
+        if (!tableMap.containsKey(fullyQualifiedTableName)) {
+            throw new NoSuchElementException(
+                    String.format("No such table found: namespace: %s, tableName: %s", namespace, tableName));
+        }
+        return (Table<K, V, M>) tableMap.get(fullyQualifiedTableName);
+    }
+
+    /**
+     * Deletes a table.
+     *
+     * @param namespace Namespace of the table.
+     * @param tableName Name of the table.
+     */
+    public void deleteTable(String namespace, String tableName) {
+        throw new UnsupportedOperationException("deleteTable unsupported for now");
+    }
+
+    /**
+     * Lists all the tables for a namespace.
+     *
+     * @param namespace Namespace for a table.
+     * @return Collection of tables.
+     */
+    public Collection<TableName> listTables(@Nullable final String namespace) {
+        return registryTable.keySet()
+                .stream()
+                .filter(tableName -> namespace == null || tableName.getNamespace().equals(namespace))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Gets the table descriptors for a particular table name.
```

Comment:
for a particular fully qualified table name.

As we are using namespace and table name for its parts and not the composed.

Reply:
thanks and yes.